pybop/costs/_likelihoods.py

```diff
@@ -235,10 +235,12 @@ def __init__(
         super().__init__(problem=log_likelihood.problem)
         self.gradient_step = gradient_step
 
-        # Store the likelihood and prior
+        # Store the likelihood, prior, update parameters and transformation
+        self.join_parameters(log_likelihood.parameters)
         self._log_likelihood = log_likelihood
-        self._parameters = self._log_likelihood.parameters
-        self._has_separable_problem = self._log_likelihood.has_separable_problem
+
+        for attr in ["transformation", "_has_separable_problem"]:
```

Comment:
To address this point, the parent class needs to be refactored, or a separate parent class needs to be created for the cost classes that compose other cost classes. I'm keen to improve this class, but it looks like it will require more changes than should be added to this PR. I'll open an action for this refactor, and this PR can proceed. Does that sound reasonable?